tools/generate-mocks/generate-mocks.ts

```diff
@@ -385,25 +435,39 @@ export class ${mockClassName} {${propertiesString}${methodsString}}
     return components;
   }
 
-  private visitTree(node: ts.Node, componentMetaData?: ComponentMetaData) {
+  private visitTree(
+    node: ts.Node,
+    componentMetaData?: ComponentMetaData,
+    assertedTypesMap?: Map<string, string>
+  ) {
     if (ts.isClassDeclaration(node)) {
-      this.visitClassDeclaration(node, componentMetaData);
+      this.visitClassDeclaration(node, componentMetaData, assertedTypesMap);
     }
     if (ts.isPropertyDeclaration(node) || ts.isSetAccessorDeclaration(node)) {
       this.visitPropertyDeclaration(node, componentMetaData);
     }
     if (ts.isMethodDeclaration(node)) {
       this.visitMethodDeclaration(node, componentMetaData);
     }
+    assertedTypesMap?.forEach((value) => {
+      if (componentMetaData.className === value) {
+        this.visitCustomElementsJSON(componentMetaData);
+      }
+    });
     ts.forEachChild(node, (node) => this.visitTree(node, componentMetaData));
   }
 
   private visitClassDeclaration(
     classDeclaration: ts.ClassDeclaration,
-    componentMetaData: ComponentMetaData
+    componentMetaData: ComponentMetaData,
+    assertedTypesMap: Map<string, string>
   ) {
     const className = classDeclaration.name.getText();
-    componentMetaData.className = className;
+
+    componentMetaData.className = assertedTypesMap?.get(className)
```

Comment:
Good catch - there was a case, but realised it could be avoided. It resulted in a small refactor in ffd909b71939bbe275e9a26074d2978d0d3a177a that should get rid of most of this optional chaining in visitTree and visitClassDecoration.